src/dj_notebook/config_helper.py

```diff
@@ -0,0 +1,124 @@
+import ast
+import os
+from pathlib import Path
+from typing import Generator, Tuple
+
+from dotenv import load_dotenv
+
+
+# taken from dotenv, which declares a similar type (but it doesn't look public...)
+# review note: the | syntax is new in python 3.10. If older pythons are generally being supported here, this should be
+# rewritten as Union[str, os.PathLike[str]]
+StrPath = str | os.PathLike[str]
+
+
+def setdefault_calls(module_path: Path) -> Generator[ast.Call, None, None]:
+    """Yields all calls to `os.environ.setdefault` within a module."""
+    with open(module_path, "r") as module_src:
+        parsed_module = ast.parse(module_src.read())
+    environ_id = None
+    for node in ast.walk(parsed_module):
+        if isinstance(node, ast.ImportFrom) and node.module == "os":
+            for name in node.names:
+                if isinstance(name, ast.alias) and name.name == "environ":
+                    environ_id = name.asname if name.asname is not None else name.name
+        if (
+            isinstance(node, ast.Call)
+            and isinstance(node.func, ast.Attribute)
+            and node.func.attr == "setdefault"
+        ):
+            if (
+                isinstance(node.func.value, ast.Attribute)
+                and node.func.value.attr == "environ"
+            ):
+                if (
+                    isinstance(node.func.value.value, ast.Name)
+                    and node.func.value.value.id == "os"
+                ):
+                    yield node
+            elif (
+                isinstance(node.func.value, ast.Name)
+                and node.func.value.id == environ_id
+            ):
+                yield node
+
+
+def is_root(path: Path) -> bool:
+    """
+    returns True if the supplied path is the root directory. This is only here because it seems clearer than
```

Comment:
Would this qualify as syntactical sugar? Regardless, it does make reading the code easier.